src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/ApplicationPropertyPage/CompoundTargetFrameworkValueProvider.cs

```diff
@@ -208,9 +212,17 @@ private async Task<string> ComputeValueAsync(ComplexTargetFramework complexTarge
             return targetFrameworkAlias;
         }
 
-        private static bool IsNetCore5OrHigher(string targetFrameworkAlias)
+        private static bool IsNetCore5OrHigher(string targetFrameworkAlias, string? targetFrameworkIdentifier)
         {
-            return targetFrameworkAlias.Contains("5.0") || targetFrameworkAlias.Contains("6.0") || targetFrameworkAlias.Contains("7.0");
+            // Converts the target framework alias to a double and checks if it's greater than or equal to 5.0.
+            // First, remove any non-numeric characters. Example: net5.0 => 5.0
+            targetFrameworkAlias = Regex.Replace(targetFrameworkAlias, "[^0-9.]", string.Empty);
+            double.TryParse(targetFrameworkAlias, out double targetFrameworkVersion);
+
+            if (targetFrameworkIdentifier is not null)
+                return targetFrameworkIdentifier.Contains(".NETCoreApp") && targetFrameworkVersion >= 5.0;
```

Comment:
Pretty sure you want `StringComparers.PropertyValues.Equals` rather than `Contains`.

The `TargetFrameworkIdentifier` should always be `.NETCoreApp`. The `TargetFrameworkMoniker` can be something like `.NETCoreApp,Version=v7.0`.

Reply:
This code was deleted on latest commit.